pimd/pim_ifchannel.c

```diff
@@ -233,10 +233,17 @@ void pim_ifchannel_delete_all(struct interface *ifp)
 
 void delete_on_noinfo(struct pim_ifchannel *ch)
 {
-	if (ch->local_ifmembership == PIM_IFMEMBERSHIP_NOINFO
-	    && ch->ifjoin_state == PIM_IFJOIN_NOINFO
-	    && ch->t_ifjoin_expiry_timer == NULL)
+	struct pim_upstream *up = ch->upstream;
+	/*
+	 * (S,G) with no active traffic, KAT expires, PPT expries,
+	 * channel state is NoInfo
+	 */
+	if (ch->local_ifmembership == PIM_IFMEMBERSHIP_NOINFO &&
+	    ch->ifjoin_state == PIM_IFJOIN_NOINFO &&
+	    (ch->t_ifjoin_expiry_timer == NULL ||
+	     (up && !pim_upstream_is_kat_running(up)))) {
```

Comment:
For (S,G) this looks good.
However this condition pim_upstream_is_kat_running will always be false, so when when join_expiry_timer is running, the code will delete the ifchannel.
Can you please check this once ?

Reply:
@routingrocks : any update on this one?